types/row_proof.go

```diff
@@ -34,6 +34,11 @@ func (rp RowProof) Validate(root []byte) error {
 	if len(rp.Proofs) != len(rp.RowRoots) {
 		return fmt.Errorf("the number of proofs %d must equal the number of row roots %d", len(rp.Proofs), len(rp.RowRoots))
 	}
+	if len(rp.Proofs) != 0 &&
+		(int64(rp.StartRow) != rp.Proofs[0].Index ||
+			int64(rp.StartRow) != rp.Proofs[len(rp.Proofs)-1].Index) {
+		return fmt.Errorf("invalid start/end row")
+	}
```

Comment:
I'm a bit lost on why this conditional is an invalid start or end row. EndRow isn't used in the conditional so how can it be deemed invalid?

Can we add a unit test in this PR?

Reply:
[fead057](https://github.com/celestiaorg/celestia-core/pull/1476/commits/fead057e432fee74772602a51fcb4185845a8986)